controllers/userController.js

```diff
@@ -11,66 +11,81 @@ const getUserInfo = async (req, res) => {
   res.status(200).send({
     payload: {
       user
-    },
+    }
   });
 };
 
 const getAllUsers = async (req, res, next) => {
   try {
-    if(accessLevelCheck(req.user.level, "admin")){
+    if (accessLevelCheck(req.user.level, 'admin')) {
       const users = await getUsers();
       res.status(200).send({
-        payload: users,
+        payload: users
       });
     } else {
       throw new AppError('Only admin can get all the users');
     }
-  }catch(error){
+  } catch (error) {
     next(new AppError(error.message));
   }
-}
+};
 
 const updateOneUser = async (req, res, next) => {
   try {
     const id = req.params.userId;
-    if((req.user._id).toString() === id || accessLevelCheck(req.user.level,'admin')){
-      const body = {...req.body}; 
-      const updatedUser = await updateUser(id, body);
-      if(updatedUser){
+    if (accessLevelCheck(req.user.level, 'admin')) {
+      const updatedUser = await updateUser(id, { ...req.body });
+      if (updatedUser) {
         res.status(200).send({
           payload: updatedUser
-        }); 
-      }else {
-        throw new AppError('user not found');
+        });
+      } else {
+        throw new AppError('User not found');
+      }
+    }
+    if (req.user._id.toString() === id) {
+      const hashedPassword = await bcrypt.hash(
```

Comment:
This will break code. Every time that user updates his information you will change password. This logic should be handled by pre hook